piker/brokers/kraken.py

```diff
@@ -162,6 +234,109 @@ async def _public(
         )
         return resproc(resp, log)
 
+    async def _private(
+        self,
+        method: str,
+        data: dict,
+        uri_path: str
+    ) -> Dict[str, Any]:
+        headers = {
+            'Content-Type': 
+                'application/x-www-form-urlencoded',
+            'API-Key':
+                self._api_key,
+            'API-Sign':
+                get_kraken_signature(uri_path, data, self._secret)
+        }
+        resp = await self._sesh.post(
+            path=f'/private/{method}',
+            data=data,
+            headers=headers,
+            timeout=float('inf')
+        )
+        return resproc(resp, log)
+
+    async def endpoint(
+        self,
+        method: str,
+        data: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        uri_path = f'/0/private/{method}'
+        data['nonce'] = str(int(1000*time.time()))
+        return await self._private(method, data, uri_path)
+
+    async def get_trades(
+        self,
+        data: Dict[str, Any] = {}
+    ) -> Dict[str, Any]:
+        data['ofs'] = 0
+        # Grab all trade history
+        # https://docs.kraken.com/rest/#operation/getTradeHistory
+        # Kraken uses 'ofs' to refer to the offset
+        while True:
+            resp = await self.endpoint('TradesHistory', data)
+            # grab the first 50 trades
+            if data['ofs'] == 0:
+                trades = resp['result']['trades']
+            # load the next 50 trades using dict constructor
+            # for speed
+            elif data['ofs'] == 50:
+                trades = dict(trades, **resp['result']['trades'])
+            # catch the end of the trades
+            elif resp['result']['trades'] == {}:
+                count = resp['result']['count']
+                break
+            # update existing dict if num trades exceeds 100
+            else: 
+                trades.update(resp['result']['trades'])
+            # increment the offset counter
+            data['ofs'] += 50
+            # To avoid exceeding API rate limit in case of a lot of trades
+            await trio.sleep(1)
+
+        # make sure you grabbed all the trades
+        assert count == len(trades.values())
+
+        return trades
+
+    async def submit_limit(
+        self,
+        oid: str,
+        symbol: str,
+        price: float,
+        action: str,
+        size: float,
+#        account: str,
```

Comment:
stale?

Reply:
the `account` argument? ye seems like it